plugins/apitracing/src/lib/FunctionHook.cpp

```diff
@@ -32,20 +32,23 @@ namespace ApiTracing
         logger->bind({{VmiCore::WRITE_TO_FILE_TAG, LOG_FILENAME}});
     }
 
-    void FunctionHook::hookFunction(addr_t moduleBaseAddress, uint64_t processCr3)
+    void FunctionHook::hookFunction(VmiCore::addr_t moduleBaseAddress,
+                                    std::shared_ptr<const VmiCore::ActiveProcessInformation> processInformation)
     {
-        auto functionEntrypoint =
-            introspectionAPI->translateUserlandSymbolToVA(moduleBaseAddress, processCr3, functionName);
+        auto functionEntrypoint = introspectionAPI->translateUserlandSymbolToVA(
+            moduleBaseAddress, processInformation->processUserDtb, functionName);
 
-        interruptEvent = pluginInterface->createBreakpoint(
-            functionEntrypoint, processCr3, VMICORE_SETUP_SAFE_MEMBER_CALLBACK(hookCallback));
+        breakpoint = pluginInterface->createBreakpoint(
+            functionEntrypoint, *processInformation, VMICORE_SETUP_SAFE_MEMBER_CALLBACK(hookCallback));
 
-        hookedProcesses.emplace_back(processCr3);
+        hookedProcessDtb = processInformation->processUserDtb;
     }
 
     BpResponse FunctionHook::hookCallback(IInterruptEvent& event)
     {
-        if (std::find(hookedProcesses.begin(), hookedProcesses.end(), event.getCr3()) != hookedProcesses.end())
+        // Sanitize cr3 in order to get rid of potential architecture specific values that do not belong to the actual
```

Comment:
Hard-coded x86 stuff here. Probably best to deal with this within the API and sanitize accordingly.

Reply:
I suppose this code can be removed since we don't need to do filtering anymore. @K-Mayer @cakeless is this assumption correct?

Reply:
That should be correct